File: backend/src/app/api/schemas/operations.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class RetentionPolicySummary(BaseModel):
    message_content_days: int
    tickets_and_notes_days: int
    audit_events_days: int
    application_logs_days: int
    backups_days: int


class DeliveryFailureSummary(BaseModel):
    message_id: UUID
    ticket_id: UUID | None
    ticket_reference: str | None
    failed_at: datetime | None
    error_code: str | None
    error_summary: str | None


class OperationalSummaryResponse(BaseModel):
    project_id: str
    generated_at: datetime
    pending_outbox: int
    stale_outbox: int
    failed_outbox: int
    oldest_pending_at: datetime | None
    unprocessed_webhook_events: int
    delivery_failures_last_24_hours: int
    messages_due_for_redaction: int
    inactive_tickets_due_for_retention: int
    last_retention_run_at: datetime | None
    retention: RetentionPolicySummary
    recent_delivery_failures: list[DeliveryFailureSummary]


class AuditEventResponse(BaseModel):
    event_id: UUID
    created_at: datetime
    actor_account_id: UUID | None
    actor_membership_id: UUID | None
    action: str
    target_type: str | None
    target_id: str | None
    outcome: str
    request_id: str | None
    metadata: dict


class AuditEventPageResponse(BaseModel):
    items: list[AuditEventResponse]
    offset: int = Field(ge=0)
    limit: int = Field(ge=1, le=100)
    total: int

File: frontend/src/api/operations.ts
import { apiRequest } from "./client";

export interface RetentionPolicySummary {
  message_content_days: number;
  tickets_and_notes_days: number;
  audit_events_days: number;
  application_logs_days: number;
  backups_days: number;
}

export interface DeliveryFailureSummary {
  message_id: string;
  ticket_id: string | null;
  ticket_reference: string | null;
  failed_at: string | null;
  error_code: string | null;
  error_summary: string | null;
}

export interface OperationalSummary {
  project_id: string;
  generated_at: string;
  pending_outbox: number;
  stale_outbox: number;
  failed_outbox: number;
  oldest_pending_at: string | null;
  unprocessed_webhook_events: number;
  delivery_failures_last_24_hours: number;
  messages_due_for_redaction: number;
  inactive_tickets_due_for_retention: number;
  last_retention_run_at: string | null;
  retention: RetentionPolicySummary;
  recent_delivery_failures: DeliveryFailureSummary[];
}

export interface AuditEvent {
  event_id: string;
  created_at: string;
  actor_account_id: string | null;
  actor_membership_id: string | null;
  action: string;
  target_type: string | null;
  target_id: string | null;
  outcome: string;
  request_id: string | null;
  metadata: Record<string, unknown>;
}

export interface AuditEventPage {
  items: AuditEvent[];
  offset: number;
  limit: number;
  total: number;
}

export function getOperationalSummary(projectId: string): Promise<OperationalSummary> {
  return apiRequest(`/api/v1/projects/${projectId}/operations/summary`);
}

export function getAuditEvents(projectId: string): Promise<AuditEventPage> {
  return apiRequest(`/api/v1/projects/${projectId}/operations/audit-events?limit=50`);
}
